src/shogun/base/SGObject.h

```diff
@@ -635,6 +635,21 @@ class CSGObject
 		return get(tag);
 	}
 
+	/**
+	 *
+	 * @param name name of the parameter
+	 * @return value of the parameter corresponding to the input name and type
+	 */
+	void run(const std::string& name) const noexcept(false)
+	{
+		Tag<bool> tag(name);
+		auto param = get_function(tag);
+		if (!any_cast<bool>(param.get_value()))
```

Comment:
Is that really a good idea? This Boolean business? Then we need to make sure that methods are actually written in a way that a false indicates a failure?

Reply:
well the issue is that we can't store void functions with any, so this was a work around. Adding support for that would involve quite a bit of redesign I think..

Reply:
I wonder: maybe this could be done with a return type that is just used for this case?

Reply:
ok let's use it like that for now, and then refactor if it causes problems.